### PR TITLE
Discontinue packages if the moderated user is single owner.

### PR DESCRIPTION
**Pull request by isoos**
#7535

### CODE REVIEW

File: app/lib/admin/actions/moderate_user.dart [hunk 1]
```diff
@@ -56,6 +62,51 @@ The active web sessions of the user will be expired.
...
+        for (final e in publishers.publishers!) {
+          final p = await publisherBackend.getPublisher(e.publisherId);
+          if (p == null) {
+            continue;
+          }
+          final members =
+              await publisherBackend.listPublisherMembers(e.publisherId);
+          if (members.length != 1) {
+            continue;
+          }
```
Comment (jonasfj) [inline app/lib/admin/actions/moderate_user.dart:92]: Will this also count blocked members?

**Review by jonasfj — APPROVED** [review]

File: app/lib/admin/actions/moderate_user.dart [hunk 2]
```diff
@@ -56,6 +62,51 @@ The active web sessions of the user will be expired.
...
       await accountBackend.updateModeratedFlag(user!.userId, valueToSet);
       user2 = await accountBackend.lookupUserById(user.userId);
+
+      if (valueToSet) {
+        await for (final p
+            in packageBackend.streamPackagesForUser(user.userId)) {
+          await withRetryTransaction(dbService, (tx) async {
+            final key = dbService.emptyKey.append(Package, id: p);
+            final pkg = await tx.lookupOrNull<Package>(key);
+            if (pkg == null || pkg.isDiscontinued || pkg.uploaderCount != 1) {
```
[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/admin/actions/moderate_user.dart:72]: > What about packages with a publisher?

Those are handled below with a separate query.

[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/admin/actions/moderate_user.dart:72]: Renaming to `streamPackagesWhereUserIsUploader`.

File: app/lib/admin/actions/moderate_user.dart (hunk 1 above)
Comment (isoos) [reply app/lib/admin/actions/moderate_user.dart]: Fixed.

**Pull request merged**